Source/Core/Common/StringUtil.cpp

```diff
@@ -369,6 +369,18 @@ std::string JoinStrings(const std::vector<std::string>& strings, const std::stri
   return joined.substr(0, joined.length() - delimiter.length());
 }
 
+std::string Trim(const std::string& str)
```

Comment:
This seems very similar to the StripSpaces function that already exists in this file. Would it be possible to use that instead?

Reply:
Hmm, I was surprised when I thought a trim function didn't exist.  Anyway, the one issue I have with `StripSpaces` is it returns empty string if there aren't any spaces.  That probably won't work for me as it's user defined (so it might not have a space in it).  The function is used all over the place, not sure I should change the functionality?

Reply:
Also, not a huge fan with the name (it implies it removes all spaces) but if it works, I'm happy to adopt it.

Reply:
I suppose I can use `StripSpaces` then check if it's empty and return the actual string

Reply:
> Anyway, the one issue I have with `StripSpaces` is it returns empty string if there aren't any spaces.

That's not what it does as far as I know. When reading the code, keep in mind that it's using `find_first_not_of`, not `find_first_of`.